docs/examples/plot_greensboro_kimber_soiling.py

```diff
@@ -0,0 +1,63 @@
+"""
+Kimber Soiling Model
+====================
+
+Examples of soiling using the Kimber model [1]_.
+
+References
+----------
+.. [1] "The Effect of Soiling on Large Grid-Connected Photovoltaic Systems
+   in California and the Southwest Region of the United States," Adrianne
+   Kimber, et al., IEEE 4th World Conference on Photovoltaic Energy
+   Conference, 2006, :doi:`10.1109/WCPEC.2006.279690`
+"""
+
+# %%
+# This example shows basic usage of pvlib's Kimber Soiling model with
+# :py:meth:`pvlib.losses.soiling_kimber`.
+#
+# The Kimber Soiling model assumes that soiling builds up at a constant rate
+# until cleaned either manually or by rain. The rain must reach a threshold to
+# clean the panels. When rains exceeds the threshold, it's assumed the earth is
+# damp for a grace period before it begins to soil again. There is a maximum
+# soiling build up that cannot be exceeded even if there's no rain or
+# manual cleaning.
+#
+# Threshold
+# ---------
+# The example shown here demonstrates how the threshold affects soiling.
+# Because soiling depends on rainfall, loading weather data is always the first
+# step.
+
+from datetime import datetime
+from matplotlib import pyplot as plt
+from pvlib.iotools import read_tmy3
+from pvlib.losses import soiling_kimber
+from pvlib.tests.conftest import DATA_DIR
```

Comment:
Is there a better way to access the data directory from the examples? I don't think it's a good idea to suggest to people that they should import anything from modules in `pvlib.tests`.

Reply:
Sure, sorry me just being lazy:

1. I can create the path to `DATA_DIR`, something like `import os;DATA_DIR=os.path.join(is.path.dirname(__file__),'..','..','pvlib','data')` except maybe using `pathlib` and pep8 formatting. This is my preference so no network calls
2. I could retrieve the actual Greensboro TMY3 file from the NSRDB by URL, but this would require a network call for every CI run, unless it's decorated with the remote network pytest thing, see #858

Reply:
This seems like it might come up with future gallery examples too.  Some packages have a function that provides access to the built-in datasets.  That way they are accessible to users without them having to monkey around with filepaths.  Examples:
- https://scikit-learn.org/stable/datasets/index.html#toy-datasets
- https://seaborn.pydata.org/generated/seaborn.load_dataset.html

What about a new function (maybe in `pvlib.tools`?) that looks like `load_dataset('greensboro-tmy')` or something?

Reply:
Hi Kevin, fo you know if there is already an open issue for this? Thanks

Reply:
See #924 :)